### PR TITLE
GG-34258 Add getting/changing the priority of compute tasks for monitoring

### PR DESCRIPTION
**Pull request by tkalkirill**
https://ggsystems.atlassian.net/browse/GG-34258

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/managers/collision/GridCollisionManager.java [hunk 1]
```diff
@@ -130,4 +129,11 @@ public void onCollision(
             });
         }
     }
+
+    /**
+     * @return Collision SPI.
+     */
+    public CollisionSpi collisionSpi() {
```
Comment (ibessonov) [inline modules/core/src/main/java/org/apache/ignite/internal/managers/collision/GridCollisionManager.java:136]: This is literally equal to `ctx.config().getCollisionSpi()`, right? Why do you need this method?

File: modules/core/src/main/java/org/apache/ignite/internal/processors/job/GridJobProcessor.java [hunk 2]
```diff
@@ -343,6 +357,15 @@ public GridJobProcessor(GridKernalContext ctx) {
 
                 return new ComputeJobView(e.getKey(), e.getValue(), state);
             });
+
+        if (!jobAlwaysActivate && ctx.collision().collisionSpi() instanceof PriorityQueueCollisionSpi) {
```
Comment (ibessonov) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/job/GridJobProcessor.java:361]: Maybe extract `ctx.collision().collisionSpi()` into a variable?

File: modules/core/src/main/java/org/apache/ignite/internal/processors/job/GridJobProcessor.java [hunk 3]
```diff
@@ -2282,4 +2311,49 @@ private JobsMap(int initCap, float loadFactor, int concurLvl) {
...
     }
+
+    /**
+     * Callback on changing task attributes.
+     *
+     * @param sesId Session ID.
+     * @param jobId Job ID.
+     * @param attrs Changed attributes.
+     */
+    public void onChangeTaskAttributes(IgniteUuid sesId, IgniteUuid jobId, Map<?, ?> attrs) {
```
Comment (ibessonov) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/job/GridJobProcessor.java:2322]: Why is it public?

File: modules/core/src/main/java/org/apache/ignite/internal/processors/task/GridTaskProcessor.java [hunk 4]
```diff
@@ -1064,7 +1062,7 @@ private void sendSessionAttributes(Map<?, ?> attrs, GridTaskSessionImpl ses)
 
                 UUID nodeId = sib.nodeId();
 
-                if (!nodeId.equals(locNodeId) && !sib.isJobDone() && !rcvrs.contains(nodeId))
+                if (!sib.isJobDone())
```
Comment (ibessonov) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/task/GridTaskProcessor.java:1065]: So, you add a job to `rcvrs` collection even if it's deployed locally

File: modules/core/src/main/java/org/apache/ignite/internal/processors/task/GridTaskProcessor.java [hunk 5]
```diff
@@ -1087,14 +1085,20 @@ private void sendSessionAttributes(Map<?, ?> attrs, GridTaskSessionImpl ses)
...
 
             UUID nodeId = sib.nodeId();
 
-            // Pair can be null if job is finished.
-            if (rcvrs.remove(nodeId)) {
+            if (locNodeId.equals(nodeId)) {
+                // Local job notification.
+                ctx.job().onChangeTaskAttributes(ses.getId(), s.getJobId(), attrs);
+            }
+            else if (rcvrs.remove(nodeId)) {
```
Comment (ibessonov) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/task/GridTaskProcessor.java:1100]: ... but don't remove it after. Am I missing something?

File: modules/core/src/main/java/org/apache/ignite/internal/processors/job/GridJobProcessor.java (hunk 3 above)
Comment (tkalkirill) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/job/GridJobProcessor.java:2322]: We need to call from class `GridTaskProcessor#sendSessionAttributes` and `GridJobProcessor#processTaskSessionRequest` and therefore `public`.

File: modules/core/src/main/java/org/apache/ignite/internal/managers/collision/GridCollisionManager.java (hunk 1 above)
Comment (tkalkirill) [reply modules/core/src/main/java/org/apache/ignite/internal/managers/collision/GridCollisionManager.java:136]: You're right, I can do that.

File: modules/core/src/main/java/org/apache/ignite/internal/processors/job/GridJobProcessor.java (hunk 2 above)
Comment (tkalkirill) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/job/GridJobProcessor.java:361]: You're right, I can do that.

File: modules/core/src/main/java/org/apache/ignite/internal/processors/task/GridTaskProcessor.java (hunk 4 above)
Comment (tkalkirill) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/task/GridTaskProcessor.java:1065]: You're right, I fix it.

File: modules/core/src/main/java/org/apache/ignite/internal/processors/task/GridTaskProcessor.java (hunk 5 above)
Comment (tkalkirill) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/task/GridTaskProcessor.java:1100]: You're right, it's bad.

File: modules/core/src/main/java/org/apache/ignite/internal/managers/collision/GridCollisionManager.java [hunk 6]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2021 GridGain Systems, Inc. and Contributors.
+ * Copyright 2019 GridGain Systems, Inc. and Contributors.
```
Comment (ibessonov) [inline modules/core/src/main/java/org/apache/ignite/internal/managers/collision/GridCollisionManager.java]: Hmm...

**Review by ibessonov — APPROVED** [review]

**Pull request merged**